src/ValidationContext.php

```diff
@@ -39,10 +39,12 @@ final class ValidationContext
 
     /**
      * @var DataSetInterface|null Data set the attribute belongs to.
-     * `null` if data set was not set with {@see setDataSet()} yet.
+     * `null` if data set was not set with {@see setIsolatedDataSet()} yet.
      */
     private ?DataSetInterface $dataSet = null;
 
+    private ?DataSetInterface $isolatedDataSet = null;
```

Comment:
Need to add PHPDoc.

Reply:
And mention the difference between `$dataSet` and `$isolatedDataSet` (in the other places as well).